drivers/fuel_gauge/charger_act81461.c

```diff
@@ -0,0 +1,270 @@
+					      uint32_t pins)
+{
+	struct act81461_gauge_data *data =
+		CONTAINER_OF(cb, struct act81461_gauge_data, gpio_int_cb);
+	k_work_submit(&data->interrupt_work);
+	LOG_DBG("pmic battery charger status change Interrupt");
+}
+
+static int act81461_charger_get_prop(const struct device *dev, struct fuel_gauge_get_property *prop)
+{
+	int rc = 0;
+	uint8_t val = 0;
+	const struct act81461_gauge_config *config = dev->config;
+
+	switch (prop->property_type) {
+	case FUEL_GAUGE_TEMPERATURE:
+		rc = i2c_reg_read_byte_dt(&config->i2c, CHG_STAT_REG, &val);
+		if (rc) {
+			break;
+		}
+		val &= BAT_TEMP_MSK;
+		switch (val) {
+		case 0 << BAT_TEMP_SHIFT: /* Below 0 (C) */
+			prop->value.temperature = DEG_C(-5);
+		case 1 << BAT_TEMP_SHIFT: /* 0 to 10 (C) */
+			prop->value.temperature = DEG_C(5);
+		case 2 << BAT_TEMP_SHIFT: /* 10 to 45 (C) */
+			prop->value.temperature = DEG_C(25);
+		case 3 << BAT_TEMP_SHIFT: /* 45 to 50 (C) */
+			prop->value.temperature = DEG_C(45);
+		case 4 << BAT_TEMP_SHIFT: /* 50 to 60 (C) */
+			prop->value.temperature = DEG_C(55);
+		case 5 << BAT_TEMP_SHIFT: /* Over 60 (C) */
+			prop->value.temperature = DEG_C(65);
+		default:
+			rc = -EIO;
+		}
+		break;
+	case FUEL_GAUGE_STATUS:
+		prop->value.fg_status = 0;
+		rc = i2c_reg_read_byte_dt(&config->i2c, CHG_STAT_REG, &val);
+		if (rc) {
+			break;
+		}
+
+		uint8_t chg_stat = val & CHG_STAT_MSK;
+
+		switch (chg_stat) {
+		case 0b000: /* Exit Charge */
+			LOG_DBG("Charger has exited charge mode");
+			prop->value.fg_status |= FUEL_GAUGE_STATUS_FLAGS_DISCHARGING;
+			break;
+		case 0b110: /* End of Charge Mode */
+			LOG_DBG("Battery end of charge has been detected");
+			prop->value.fg_status |= FUEL_GAUGE_STATUS_FLAGS_FULLY_CHARGED;
+			break;
+
+		case 0b111: /* Fault Mode */
+			LOG_DBG("Battery fault detected");
+			/* TODO: Add timer
+			 * LOG_ERR("Battery fault has been detected for %d
+			 * minutes",					pmic_battery_fault_count *
+			 * NO_RECOVER_MSG_INTERVAL_SECONDS /
+			 * 60);
+			 * LOG_ERR("Disconnect and reattach battery to
+			 * clear this fault");
+			 */
+			prop->value.fg_status |= FUEL_GAUGE_STATUS_FLAGS_TERMINATE_CHARGE_ALARM;
+			break;
+		case 0b010: /* VBAT SHORT Mode */
+			LOG_DBG("Charger is in VBAT SHORT mode");
+			prop->value.fg_status |= FUEL_GAUGE_STATUS_FLAGS_TERMINATE_CHARGE_ALARM;
+			break;
+		case 0b001: /* Reset Mode */
+			LOG_DBG("Charger is in reset mode");
+			break;
+		case 0b011: /* VBAT PRECOND Mode */
+			LOG_DBG("Charger is in VBAT PRECOND mode");
+			break;
+		case 0b100: /* Fast Charge CC loop on */
+		case 0b101: /* Fast Charge CC loop off & CV loop on */
+			LOG_DBG("Battery is in FAST charge mode - charging");
+		default:
+			break;
+		}
+
+		rc = i2c_reg_read_byte_dt(&config->i2c, BAT_STAT_REG, &val);
+
+		if (rc) {
+			break;
+		}
+
+		if (val & SUSPEND) {
+			prop->value.fg_status |= FUEL_GAUGE_STATUS_FLAGS_OVER_TEMP_ALARM;
+		}
+
+		break;
```

Comment:
Suggest putting this blank line after the break instead of before.

Also suggest adding blank line after other breaks that don't already have one (for consistency)